From b97920910c2a6a5dab6c80dd49fcf781f01de5af Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:59 -0500
Subject: [drm] radeon/cik: use a separate counter for CP init timeout

Message-id: <1418852380-13061-49-git-send-email-rclark@redhat.com>
Patchwork-id: 102180
O-Subject: [RHEL7 drm 48/89] drm/radeon/cik: use a separate counter for CP init timeout
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit 127e837f479a2fa68f1cca05be739afb3cd3f8f0

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Tue Sep 23 10:20:13 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:20:02 2014 -0500

    drm/radeon/cik: use a separate counter for CP init timeout

    commit 370ce45b5986118fa496dddbcd7039e1aa1a418f upstream.

    Otherwise we may fail to init the second compute ring.

    Noticed-by: Christian Koenig <christian.koenig@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/cik.c b/drivers/gpu/drm/radeon/cik.c
index 78b4a27..5ea01de 100644
--- a/drivers/gpu/drm/radeon/cik.c
+++ b/drivers/gpu/drm/radeon/cik.c
@@ -4489,7 +4489,7 @@ struct bonaire_mqd
  */
 static int cik_cp_compute_resume(struct radeon_device *rdev)
 {
- int r, i, idx;
+ int r, i, j, idx;
  u32 tmp;
  bool use_doorbell = true;
  u64 hqd_gpu_addr;
@@ -4608,7 +4608,7 @@ static int cik_cp_compute_resume(struct radeon_device *rdev)
   mqd->queue_state.cp_hqd_pq_wptr= 0;
   if (RREG32(CP_HQD_ACTIVE) & 1) {
    WREG32(CP_HQD_DEQUEUE_REQUEST, 1);
-   for (i = 0; i < rdev->usec_timeout; i++) {
+   for (j = 0; j < rdev->usec_timeout; j++) {
     if (!(RREG32(CP_HQD_ACTIVE) & 1))
      break;
     udelay(1);
-- 
1.7.1